syntax = "proto3";

package grpc;

option go_package = "./shopping-expenses.v1;shopping";

service ShoppingService {
  rpc Add (AddRequest) returns (AddResponse);
  rpc List (ListRequest) returns (ListResponse);
}

message AddRequest {
  int32 tg_id = 1;
  string qr_info = 2;
}

message AddResponse {
  int32 count = 1;
}

message ListRequest {
  int32 tg_id = 1;
  string begin_time = 2;
  string end_time = 3;
}

message ListResponse {
  repeated Purchase purchases = 1;
}

message Purchase {
  int32 id = 1;
  int32 user_id = 2;
  string name = 3;
  string market = 4;
  int32 price = 5;
  string category = 6;
  int32 quantity = 7;
  string barcode = 8;
}